ts/WoltLabSuite/WebComponent/woltlab-core-pagination.ts

```diff
@@ -0,0 +1,318 @@
+/**
+ * The `<woltlab-core-pagination>` creates a pagination.
+ * Usage: `<woltlab-core-pagination page="1" count="10" url="https://www.woltlab.com"></woltlab-core-pagination>`
+ *
+ * @author Marcel Werk
+ * @copyright 2001-2022 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ */
+
+{
+  let mediaQuery: MediaQueryList;
+  const getMediaQueryScreenXs = (): MediaQueryList => {
+    if (mediaQuery === undefined) {
+      mediaQuery = window.matchMedia("(max-width: 544px)");
+    }
+
+    return mediaQuery;
+  };
+
+  interface WoltlabCorePaginationEventMap {
+    jumpToPage: CustomEvent;
+    switchPage: CustomEvent<number>;
+  }
+
+  class WoltlabCorePaginationElement extends HTMLElement {
+    readonly #className = "pagination";
+
+    connectedCallback() {
+      this.#render();
+
+      getMediaQueryScreenXs().addEventListener("change", () => this.#render());
+    }
+
+    #render(): void {
+      this.innerHTML = "";
+
+      if (this.count < 2) {
+        return;
+      }
+
+      this.classList.add(`${this.#className}__wrapper`);
+
+      const nav = this.#getNavElement();
+      this.append(nav);
+
+      const previousLinkElement = this.#getPreviousLinkElement();
+      if (previousLinkElement) {
+        nav.append(previousLinkElement);
+      }
+
+      const ul = document.createElement("ul");
+      ul.classList.add(this.#className + "__list");
```

Comment:
```suggestion
      ul.classList.add(`${this.#className}__list);
```
Injecting `#className` is currently performed inconsistently. A template string is the more readable variant.